File: Dataset/Leetcode/train/70/438.py
class Solution:
    def XXX(self, n: int) -> int:
        if n==1:
            return 1
        if n==2:
            return 2
        dp=[0]*(n+1)
        dp[1],dp[2]=1,2
        for i in range(3,n+1):
            dp[i]=dp[i-1]+dp[i-2]
        return dp[n]

File: Dataset/Leetcode/valid/118/438.cpp
 class Solution {
public:
    vector<vector<int>> XXX(int numRows) {
        vector<vector<int>> dp(numRows, vector<int>(numRows));
        vector<vector<int>> res(numRows);
        for(int i=0; i<numRows; i++){
            for(int j=0; j<=i; j++){
                if(j==0 || j==i) dp[i][j] = 1;
                else
                    dp[i][j] = dp[i-1][j-1] + dp[i-1][j];
                res[i].push_back(dp[i][j]);
            }
        }
        return res;
    }
};

